File: docs-old/datatools/codesnippet/CSharp/save-data-with-the-tableadapter-dbdirect-methods_1.cs
        private void InsertButton_Click(object sender, EventArgs e)
        {
            Int32 newRegionID = 5;
            String newRegionDescription = "NorthEastern";

            try
            {
                regionTableAdapter1.Insert(newRegionID, newRegionDescription);
            }
            catch (Exception ex)
            {
                MessageBox.Show("Insert Failed");
            }
            RefreshDataset();
        }


        private void RefreshDataset()
        {
            this.regionTableAdapter1.Fill(this.northwindDataSet1.Region);
        }

File: docs/data-tools/codesnippet/VisualBasic/save-data-with-the-tableadapter-dbdirect-methods_1.vb
    Private Sub InsertButton_Click() Handles InsertButton.Click

        Dim newRegionID As Integer = 5
        Dim newRegionDescription As String = "NorthEastern"

        Try
            RegionTableAdapter1.Insert(newRegionID, newRegionDescription)

        Catch ex As Exception
            MessageBox.Show("Insert Failed")
        End Try

        RefreshDataset()
    End Sub


    Private Sub RefreshDataset()
        Me.RegionTableAdapter1.Fill(Me.NorthwindDataSet1._Region)
    End Sub